### PR TITLE
MESH-2145/allow-the-owner-to-reject-proposal

### PR DESCRIPTION
**Pull request by HenriqueNogara**
Change MS reject proposal logic. If no one else has voted, allows the owner of the proposal to reject it immediately. 

### new features
 - If no one else has voted, allows the owner of the proposal to reject it immediately.
### new events
 - Adds the `ProposalFailedToExecute` event to the multsig pallet;

### CODE REVIEW

File: pallets/common/src/traits/multisig.rs [hunk 1]
```diff
@@ -70,6 +70,9 @@ decl_event!(
         ProposalExecutionFailed(DispatchError),
         /// Scheduling of proposal fails.
         SchedulingFailed(DispatchError),
+        /// Event emitted when there's an error in proposal execution
+        /// Arguments: caller DID, multisig, proposal ID, error.
+        PoposalFailedToExecute(IdentityId, AccountId, u64, DispatchError),
```
Comment (HenriqueNogara) [inline pallets/common/src/traits/multisig.rs]: This was requested on MESH-2118, but I'm not sure we need it. From what I see, if a proposal fails, two events are emitted `ProposalExecutionFailed` and `ProposalExecuted` the second event has the information requested.

Comment (Neopallium) [reply pallets/common/src/traits/multisig.rs]: I don't think we need this event then.  But we should improve the description of the `ProposalExecuted` event, to explain that it is emitted for both success and failure.  @F-OBrien Do the two events provide what is needed?

File: pallets/multisig/src/lib.rs [hunk 2]
```diff
@@ -927,16 +932,12 @@ impl<T: Config> Module<T> {
...
-                            multisig.clone(),
-                            proposal_id,
-                        ));
-                    }
+                    proposal_details.status = ProposalStatus::Rejected;
+                    Self::deposit_event(RawEvent::ProposalRejected(
+                        current_did,
+                        multisig.clone(),
+                        proposal_id,
+                    ));
```
Comment (Neopallium) [inline pallets/multisig/src/lib.rs]: This would make any rejection stop the proposal, even if there are more approvals.

**Review by Neopallium — CHANGES_REQUESTED** [review]
A single reject vote shouldn't put the proposal into the `Rejected` state.

**Review by Neopallium — APPROVED** [review]

File: pallets/multisig/src/lib.rs [hunk 3]
```diff
@@ -884,7 +884,12 @@ impl<T: Config> Module<T> {
                 }
                 Err(e) => {
                     update_proposal_status(ProposalStatus::ExecutionFailed);
-                    Self::deposit_event(RawEvent::ProposalExecutionFailed(e.error));
+                    Self::deposit_event(RawEvent::ProposalFailedToExecute(
```
Comment (Neopallium) [inline pallets/multisig/src/lib.rs]: I think we should still emit the old event, otherwise we will need to force an upgrade to SubQuery.

**Review by Neopallium — APPROVED** [review]

**Review by adamdossa — APPROVED** [review]

**Pull request merged**